contracts/Gauge.sol

```diff
@@ -0,0 +1,451 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "./interfaces/IPool.sol";
+import "./interfaces/IGauge.sol";
+import "./CommunityFactory.sol";
+import "./interfaces/ICommunity.sol";
+import "./ERC20Helper.sol";
+import "./NUTToken.sol";
+import "./NutPower.sol";
+
+contract Gauge is IGauge, Ownable, ERC20Helper, ReentrancyGuard {
+
+    using SafeMath for uint256;
+    using SafeMath for uint16;
+
+    uint16 constant CONSTANT_10000 = 10000;
+
+    struct User {
+        bool hasDeposited;
+        uint256 amount; // np amount
+        uint256 nutAvailable;
+        uint256 nutDebt;
+        uint256 cTokenAvailable;
+        uint256 cTokenDebt;
+    }
+
+    struct GaugeMeta {
+        bool hasCreated;
+        address community;
+        address factory;
+        address cToken;
+        uint256 cTokenAcc;
+        uint256 cTokenRevenue;
+        uint256 lastCTokenRevenue;
+        uint256 totalLockedNP;
+        mapping (address => User) users;
+    }
+
+    // define the NUT distribution to 3 parts
+    struct DistributionRatio {
+        uint16 community;
+        uint16 poolFactory;
+        uint16 user;
+    }
+    
+    // Nutbox Power addresss
+    address immutable NP;
+    // NUT address
+    address immutable NUT;
+    // Only Nutbox committee can set this. Communities which enabled this function in their pools,
+    // part of the cToken rewards(dappToolsRatio / 10000 of total rewards) will be transfered
+    // to gauge contract when user harvest their rewards, those rewards and NUTs would be
+    // distributed based on the NP locked by user.
+    uint16 public gaugeRatio;
+    // total reward nut per block, can be reset by Nutbox DAO
+    uint256 public rewardNUTPerBlock;
+    // last nut reward block
+    uint256 private lastRewardBlock;
+
+    // nutAcc means how many nut will 1 NP locked earn, it departed by community/poolFactory/user
+    uint256 private userNutAcc;
+    uint256 private poolFactoryNutAcc;
+    uint256 private communityNutAcc;
+
+    uint256 public totalNPLocked;
+
+    // can be rest by Nutbox DAO, total of the ratios should be 10000
```

Comment:
What's the type of Nutbox DAO? An address or the committee address? If is committee address, we can change it to `Nutbox committee`